builtin/mainmenu/settings/dlg_file_browser.lua

```diff
@@ -0,0 +1,160 @@
+-- Minetest
+-- Copyright (C) 2020 Jean-Patrick Guerrero <jeanpatrick.guerrero@gmail.com>
+-- Copyright (C) 2020 EvidenceBKidscode
+-- Copyright (C) 2023 ROllerozxa
+--
+-- This program is free software; you can redistribute it and/or modify
+-- it under the terms of the GNU Lesser General Public License as published by
+-- the Free Software Foundation; either version 2.1 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Lesser General Public License for more details.
+--
+-- You should have received a copy of the GNU Lesser General Public License along
+-- with this program; if not, write to the Free Software Foundation, Inc.,
+-- 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local PATH = os.getenv("HOME") -- Linux & similar
+		or os.getenv("HOMEDRIVE") .. os.getenv("HOMEPATH") -- Windows
+		or core.get_user_path() -- If nothing else works
+
+tabdata = {}
+
+local function texture(filename)
+	return core.formspec_escape(defaulttexturedir .. filename .. ".png")
+end
+
+local function get_dirs()
+	local dirs
+
+	if PATH == "\\" then -- Windows root dir
+		dirs = core.get_dir_list(os.getenv("HOMEDRIVE"))
+	else
+		dirs = core.get_dir_list(PATH)
+	end
+
+	local new = {}
+	for _, f in ipairs(dirs) do
+		if f:sub(1,1) ~= "." then
+			new[#new + 1] = core.formspec_escape(f)
+		end
+	end
+
+	table.sort(new)
+	return new
+end
+
+local function make_fs(dialogdata)
+	local dirs = get_dirs()
+
+	local _dirs = ""
+
+	if not next(dirs) then
+		_dirs = "2,,"
+	end
+
+	for _, f in ipairs(dirs) do
+		local is_file = not core.is_dir(PATH .. DIR_DELIM .. f)
```

Comment:
The filename in `f` is already formspec-escaped.

Reply:
Uh... yeah?

Reply:
The point is escaped strings are not valid filesystem paths

Reply:
E.g. if you have a folder called `$hello$`, it's shown as a file.

Reply:
this is not fixed.